src/vs/workbench/contrib/inlineChat/electron-sandbox/inlineChatActions.ts

```diff
@@ -0,0 +1,88 @@
+import { KeybindingWeight } from 'vs/platform/keybinding/common/keybindingsRegistry';
+import { InlineChatController, InlineChatRunOptions } from 'vs/workbench/contrib/inlineChat/browser/inlineChatController';
+import { AbstractInlineChatAction } from 'vs/workbench/contrib/inlineChat/browser/inlineChatActions';
+import { LOCALIZED_START_INLINE_CHAT_STRING, START_INLINE_CHAT } from '../browser/inlineChatActions';
+import { disposableTimeout } from 'vs/base/common/async';
+import { EditorContextKeys } from 'vs/editor/common/editorContextKeys';
+import { ICommandService } from 'vs/platform/commands/common/commands';
+import { IKeybindingService } from 'vs/platform/keybinding/common/keybinding';
+import { StartVoiceChatAction, StopListeningAction } from 'vs/workbench/contrib/chat/electron-sandbox/actions/voiceChatActions';
+import { CTX_INLINE_CHAT_HAS_PROVIDER, InlineChatConfigKeys } from 'vs/workbench/contrib/inlineChat/common/inlineChat';
+import { IConfigurationService } from 'vs/platform/configuration/common/configuration';
+import { ISpeechService } from 'vs/workbench/contrib/speech/common/speechService';
+
+
+export class StartSessionAction extends EditorAction2 {
+
+	constructor() {
+		super({
+			id: 'inlineChat.start',
+			title: { value: LOCALIZED_START_INLINE_CHAT_STRING, original: 'Start Inline Chat' },
+			category: AbstractInlineChatAction.category,
+			f1: true,
+			precondition: ContextKeyExpr.and(CTX_INLINE_CHAT_HAS_PROVIDER, EditorContextKeys.writable),
+			keybinding: {
+				when: EditorContextKeys.focus,
+				weight: KeybindingWeight.WorkbenchContrib,
+				primary: KeyMod.CtrlCmd | KeyCode.KeyI,
+				secondary: [KeyChord(KeyMod.CtrlCmd | KeyCode.KeyK, KeyCode.KeyI)],
+			},
+			icon: START_INLINE_CHAT
+		});
+	}
+
+
+	override runEditorCommand(accessor: ServicesAccessor, editor: ICodeEditor, ..._args: any[]) {
+
+		const configService = accessor.get(IConfigurationService);
+		const speechService = accessor.get(ISpeechService);
+		const keybindingService = accessor.get(IKeybindingService);
+		const commandService = accessor.get(ICommandService);
+
+		if (configService.getValue<boolean>(InlineChatConfigKeys.HoldToSpeech) // enabled
+			&& speechService.hasSpeechProvider  // possible
+			&& keybindingService.enableKeybindingHoldMode(this.desc.id) // holding keys
+		) {
+
+
+			let listening = false;
+			const handle = disposableTimeout(() => {
+				// start VOICE input
+				commandService.executeCommand(StartVoiceChatAction.ID);
+				listening = true;
+			}, 100);
+
+			const listener = dom.addDisposableListener(
+				dom.getWindow(editor.getDomNode()),
+				dom.EventType.KEY_UP,
+				(_e: KeyboardEvent) => {
+					listener.dispose(); // Event.once
+					if (listening) {
+						commandService.executeCommand(StopListeningAction.ID).finally(() => {
+							InlineChatController.get(editor)?.acceptInput();
+						});
+					} else {
+						handle.dispose();
+					}
+				}
+			);
```

Comment:
in future: I wonder if it would be better if the keybinding service fired an event that the holding-mode command registered for rather than setting up its own key_up DOM event listenter.

Reply:
Yeah, I was thinking to make enableHoldMode return a promise that resolves when hold more is over